workflows/prognostic_c48_run/runtime/steppers/reservoir.py

```diff
@@ -0,0 +1,233 @@
+import cftime
+import dataclasses
+from datetime import timedelta
+import pandas as pd
+from typing import Optional, MutableMapping, Hashable, Sequence, Union, cast
+import xarray as xr
+
+import fv3fit
+from fv3fit._shared import put_dir
+from fv3fit._shared.halos import append_halos_using_mpi
+from fv3fit.reservoir.model import HybridReservoirComputingModel
+from fv3fit.reservoir.adapters import ReservoirDatasetAdapter
+
+
+ReservoirAdapter = Union[ReservoirDatasetAdapter, HybridReservoirComputingModel]
+
+
+@dataclasses.dataclass
+class ReservoirConfig:
+    """
+    Reservoir model configuration.
+
+    Attributes:
+        model: URL to the global hybrid RC model
+    """
+
+    models: Sequence[str]
+    synchronize_steps: int = 1
+    reservoir_timestep: str = "3h"  # TODO: Could this be inferred?
+
+
+class _FiniteStateMachine:
+
+    INCREMENT = "increment"
+    PREDICT = "predict"
+
+    def __init__(self, init_time: Optional[cftime.DatetimeJulian] = None) -> None:
+        self._last_called = None
+        self._init_time = init_time
+        self._num_increments_completed = 0
+
+    @property
+    def init_time(self):
+        return self._init_time
+
+    @property
+    def completed_increments(self):
+        return self._num_increments_completed
+
+    def to_incremented(self):
+        # incrementing allowed anytime, e.g., synchronizing
+        self._last_called = self.INCREMENT
+        self._num_increments_completed += 1
+
+    def to_predicted(self):
+        # predict only allowed after increment has been called
+        if self._last_called != self.INCREMENT:
+            raise ValueError("Must call increment before next prediction")
+        self._last_called = self.PREDICT
+
+    def set_init_time(self, time: cftime.DatetimeJulian):
+        self._init_time = time
+
+    def __call__(self, state: str):
```

Comment:
Just curious- why this pattern (calling with hard coded options) rather than directly calling `to_incremented()` etc.?

Reply:
This is to align with the defined `Stepper` protocol  defined in `runtime.steppers.stepper`